pyproject.toml

```diff
@@ -21,15 +21,23 @@ packages = [{ include = "qibo_cloud_backends", from = "src" }]
 
 [tool.poetry.dependencies]
 python = ">=3.9,<3.13"
-qibo = "^0.2.11"
-qibo_client = "^0.0.9"
-qiskit_ibm_runtime = ">=0.17"
-qiskit_ibm_provider = ">=0.8.0"
+qibo = "^0.2.12"
+qibo_client = "^0.0.10"
+qiskit_ibm_runtime = { version = "^0.17", optional = true }
+qiskit_ibm_provider = { version = "^0.11", optional = true }
```

Comment:
Do you still need both?

Reply:
I.e. the `qiskit_ibm_provider` is deprecated in favor of the `qiskit_ibm_runtime`. Since you have the latter, do you still need the former?

Reply:
We are importing right now the provider from `qiskit_ibm_provider`:

https://github.com/qiboteam/qibo-cloud-backends/blob/ee8b7352d932669565bd76c2df51f5e6e842e361/src/qibo_cloud_backends/qiskit_client.py#L8C1-L8C44

If `qiskit_ibm_runtime` provides the same provider interface we can switch it out indeed.

Reply:
It actually has `fake_providers` https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/fake_provider, which would be quite useful for testing, not sure about real providers though...

Reply:
Maybe you should upgrade to use the `SamplerV2`, as described on the main page https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/runtime_service

(the documentation is not great, but I'm not sure we can complain that much...)

Maybe, the best is just following the guide for executing a circuit in a batch: https://docs.quantum.ibm.com/guides/run-jobs-batch